codalab/rest/util.py

```diff
@@ -182,15 +182,18 @@ def get_group_info(group_spec, need_admin, access_all_groups=False):
     Resolve |group_spec| and return the associated group_info.
     """
     user_id = request.user.user_id
+    is_root_user = (user_id == local.model.root_user_id)
 
     # If we're root, then we can access any group.
-    if user_id == local.model.root_user_id or access_all_groups:
-        user_id = None
-
-    group_info = unique_group(local.model, group_spec, user_id)
+    if is_root_user or access_all_groups:
```

Comment:
This section looks pretty tricky...can you write more documentation about what's going on?